### ISSUE TITLE
Support flint 3.0.0

### ISSUE THREAD

**Issue by chenrui333** [fredrik-johansson/arb#453]
The latest release, 2.23.0, failed to build against flint 3.0.0

<details>
<summary>error build log</summary>

```
  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpz_extras.h:208:1: error: static declaration of 'fmpz_ui_pow_ui' follows non-static declaration
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/agm.c:12In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/acosh.cIn file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/addmul.c:12:
  In file included from fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e)/tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb.h
  :^22
  :
  :
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb.h:22:
  In file included from :/tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/arf.h12::
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb.h:22:
  In file included from In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/arf.h/tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/arf.h::2727:
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpr.h:30:
  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpz_extras.h:27:
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpr.h:30:
  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpz_extras.h:208:1/opt/homebrew/include/flint/fmpz.h:422:6208: :1: :
  error: :In file included from  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpr.herror: :static declaration of 'fmpz_ui_pow_ui' follows non-static declaration30note: :
  previous declaration is here/tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpz_extras.h
  :208static declaration of 'fmpz_ui_pow_ui' follows non-static declaration:

  void fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e);
       ^fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e)
  ^
  fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e)1
  ^: error: static declaration of 'fmpz_ui_pow_ui' follows non-static declaration

  fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e)
  ^
  /opt/homebrew/include/flint/fmpz.h:422:6: note: previous declaration is here
  void fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e);
       ^
  /opt/homebrew/include/flint/fmpz.h:422:6: note: previous declaration is here

  void fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e);
       ^
  /opt/homebrew/include/flint/fmpz.h:422:6: note: previous declaration is here
  void fmpz_ui_pow_ui(fmpz_t x, ulong b, ulong e);
       ^
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/acos.c:12:
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb.h:22:
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/arf.h:27:
  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/fmpr.h:360:9: error: 'count_trailing_zeros' is deprecated. Use 'flint_ctz' instead.
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/addmul.c:12:
          count_trailing_zeros(b, c);
          ^
  /opt/homebrew/include/flint/longlong.h:37:30: note: expanded from macro 'count_trailing_zeros'
  #define count_trailing_zeros _Pragma("GCC error \"'count_trailing_zeros' is deprecated. Use 'flint_ctz' instead.\"")
                               ^
  <scratch space>:87:6: note: expanded from here
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/agm.c:12:
   generated.
  7 errors generated.
  make[2]: *** [CMakeFiles/arb.dir/acb/acos.c.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  make[2]: *** [CMakeFiles/arb.dir/acb/addmul.c.o] Error 1
  make[2]: *** [CMakeFiles/arb.dir/acb/agm.c.o] Error 1
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb/acosh.c:12:
  In file included from /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/acb.h:23:
  /tmp/arb-20231021-6179-kjq3zn/arb-2.23.0/arb.h:1041:13: error: 'count_leading_zeros' is deprecated. Use 'flint_clz' instead.
              count_leading_zeros(bits, t);
              ^
  /opt/homebrew/include/flint/longlong.h:36:29: note: expanded from macro 'count_leading_zeros'
  #define count_leading_zeros _Pragma("GCC error \"'count_leading_zeros' is deprecated. Use 'flint_clz' instead.\"")
                              ^
  <scratch space>:164:6: note: expanded from here
   GCC error "'count_leading_zeros' is deprecated. Use 'flint_clz' instead."
       ^
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/6598185571/job/17925927595
relates to Homebrew/homebrew-core#151985

**Comment by oscarbenjamin** [comment]
Arb has been merged into Flint 3.0.0 so it is not necessary to build Arb as a separate library after building Flint 3.0.0.

**Comment by chenrui333** [comment]
that is good now, I am gonna deprecate the library then, thanks for the info.
